core-bundle/src/Resources/contao/dca/tl_module.php

```diff
@@ -185,6 +185,12 @@
 			'eval'                    => array('helpwizard'=>true, 'chosen'=>true, 'submitOnChange'=>true, 'tl_class'=>'w50'),
 			'sql'                     => "varchar(64) NOT NULL default 'navigation'"
 		),
+		'menuAlias' => array(
+			'exclude'   => true,
+			'inputType' => 'text',
+			'eval'      => array('maxlength' => 255, 'tl_class' => 'w50 clr', 'unique' => true),
```

Comment:
Thinking about the `menuAlias`, I would suggest making the alias non-unique and filtering for the current theme in the menu provider, i.e., `$moduleAdapter->findBy(['menuAlias' => $name, 'pid' => $themeId])`.

This allows easy switching of themes in the front-end. But at the same time, this would disallow using navigation modules not defined in the current theme.

Reply:
What about finding any with the given alias and preferring the one from the current theme if there are multiples and one matches the `$themeId`?

On that note: should we autogenerate aliases when saving like everywhere else?

Reply:
👎 for autogenerating the alias. Because if I name the module "Navigation - Main navigation" I don't want the alias to be sth like "navigation___main_navigation". Also, the alias really is not required, if you don't want to use the Knp Menu Provider / Twig integration.

Reply:
> " I don't want the alias to be sth like "navigation___main_navigation"

The alias could just be generated from the module type + id: `custom_navigation_123`.

Reply:
This format, however, looks good. But still, whenever an alias is generated in Contao, it is, because it's necessary. In this case, however, the feature is totally optional.

Reply:
Alias generation added in 305158e77b525a7fb4fbf8372e1d2c5b222f726e.

Reply:
Ambiguous menu alias and automatic theme switching added in e0ef24cc5ce79c80d004c7a01c7525df9f557160.